src/icrar/leap-accelerate/model/cpu/MetaData.cc

```diff
@@ -118,45 +118,47 @@ namespace cpu
         m_avg_data = Eigen::MatrixXcd::Zero(ms.GetNumBaselines(), ms.GetNumPols());
         LOG(info) << "avg_data: " << memory_amount(m_avg_data.size() * sizeof(std::complex<double>));
 
+        auto fg = ms.GetFlaggedBaselines();
+
+        auto uvwShape = msmc->uvw().getColumn().shape();
+        auto uvSlice = casacore::Slicer(
+            casacore::IPosition(2,0,0),
+            casacore::IPosition(2,1,uvwShape[1]),
+            casacore::IPosition(2,1,1));
+        casacore::Array<double> uv = msmc->uvw().getColumn()(uvSlice);
```

Comment:
I posted a similar comment about how you should switch from reading full columns to reading subsets of the data. Also I think `uvwShape[1]` should be the number of rows of the MS, making the first `getColumn` call unnecessary.

Since here you are reading a slice of the cells of the whole column you should be able to use `getColumn(Slicer)` here rather than the more complex `getColumnRange(Slicer, Slicer)` that I suggested for the other use case.

Reply:
I'm not quite sure how to rearrange this one

Reply:
I don't seem to use uv yet until short baseline filtering